sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -634,4 +634,28 @@ class CachedTableSuite extends QueryTest with SQLTestUtils with SharedSQLContext
       assert(getNumInMemoryRelations(cachedPlan2) == 4)
     }
   }
+
+  test("refreshByPath should refresh all cached plans with the specified path") {
+    def f(path: String, spark: SparkSession, dataCount: Int): DataFrame = {
+      spark.catalog.refreshByPath(path)
+      val data = spark.read.parquet(path)
+      assert(data.count == dataCount)
+      val df = data.filter("id > 10")
+      df.cache
+      assert(df.count == dataCount - 11)
+      val df1 = df.filter("id > 11")
+      df1.cache
+      assert(df1.count == dataCount - 12)
+      df1
```

Comment:
The function is called twice. So actually it is meant to refresh the cache in first call. Since I will change the test to what you suggested https://github.com/apache/spark/pull/17064#discussion_r103530085, we can get rid of this confusing.